### ISSUE TITLE
autocal.diagnose prints warnings in tests on py38

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#179]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/179) by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

When I run tests I get warnings like

```
'c' argument looks like a single numeric RGB or RGBA sequence, which should be avoided as value-mapping will have precedence in case its length matches with 'x' & 'y'.  Please use a 2-D array with a single row if you really want to specify the same RGB or RGBA value for all points.
'c' argument looks like a single numeric RGB or RGBA sequence, which should be avoided as value-mapping will have precedence in case its length matches with 'x' & 'y'.  Please use a 2-D array with a single row if you really want to specify the same RGB or RGBA value for all points.
```

‌

They go away if you comment out the two calls to diagnose here:

[https://bitbucket.org/joe\_fowler/mass/src/353fe895dc990b3aaa8e566d0a11cd9dd6136da8/src/mass/calibration/test\_algorithms.py#lines-156](https://bitbucket.org/joe_fowler/mass/src/353fe895dc990b3aaa8e566d0a11cd9dd6136da8/src/mass/calibration/test_algorithms.py#lines-156)

‌

I tried commenting out all the lines that plot or scatter and have color arguments in the implementation of diagnose and couldn’t get the warnings to go away.

implementation here: [https://bitbucket.org/joe\_fowler/mass/src/353fe895dc990b3aaa8e566d0a11cd9dd6136da8/src/mass/calibration/algorithms.py#lines-319](https://bitbucket.org/joe_fowler/mass/src/353fe895dc990b3aaa8e566d0a11cd9dd6136da8/src/mass/calibration/algorithms.py#lines-319)

‌

For now I’m going to change the matplotlib log level to supress the warnings in the tests. See here:

[https://bitbucket.org/joe\_fowler/mass/src/528eae4c3f8fd2067a3397e4ce55ce47857e4873/runtests.py#lines-11](https://bitbucket.org/joe_fowler/mass/src/528eae4c3f8fd2067a3397e4ce55ce47857e4873/runtests.py#lines-11)

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Fix a matplotlib warning

I cannot reproduce #179, but if I understand the warning, this should be the solution. Fixes #179.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.7 (automated comment)
